pkg/cli/admin/upgrade/status/mco/layered_node_state.go

```diff
@@ -228,39 +240,35 @@ func isNodeMCDState(node *corev1.Node, state string) bool {
 	return dstate == state
 }
 
-func checkNodeReady(node *corev1.Node) error {
+func checkNodeReady(l *LayeredNodeState) error {
+	node := l.node
 	for i := range node.Status.Conditions {
 		cond := &node.Status.Conditions[i]
 		// We consider the node for scheduling only when its:
 		// - NodeReady condition status is ConditionTrue,
 		// - NodeDiskPressure condition status is ConditionFalse,
 		// - NodeNetworkUnavailable condition status is ConditionFalse.
 		if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting NotReady=%v with message=%q", cond.Status, cond.Message)
 			return fmt.Errorf("node %s is reporting NotReady=%v", node.Name, cond.Status)
 		}
 		if cond.Type == corev1.NodeDiskPressure && cond.Status != corev1.ConditionFalse {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting OutOfDisk=%v with message=%q", cond.Status, cond.Message)
 			return fmt.Errorf("node %s is reporting OutOfDisk=%v", node.Name, cond.Status)
 		}
 		if cond.Type == corev1.NodeNetworkUnavailable && cond.Status != corev1.ConditionFalse {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting NetworkUnavailable=%v with message=%q", cond.Status, cond.Message)
 			return fmt.Errorf("node %s is reporting NetworkUnavailable=%v", node.Name, cond.Status)
 		}
 	}
 	// Ignore nodes that are marked unschedulable
 	if node.Spec.Unschedulable {
+		l.ReasonOfUnavailability = fmt.Sprintf("node is reporting Unschedulable")
```

Comment:
I'd prefer "Node is marked unschedulable", and also I think this should not be an insight, just a message displayed in the node table.